engine/classes/ElggEntity.php

```diff
@@ -1068,6 +1068,29 @@ public function canEdit($user_guid = 0) {
 	}
 
 	/**
+	 * Can a user delete this entity?
+	 *
+	 * @tip Can be overridden by registering for the permissions_check:delete plugin hook.
+	 *
+	 * @param int $user_guid The user GUID, optionally (default: logged in user)
+	 *
+	 * @return bool Whether this entity is deletable by the given user.
+	 * @see elgg_set_ignore_access()
+	 */
+	public function canDelete($user_guid = 0) {
+		$user_guid = (int)$user_guid;
+		$user = get_entity($user_guid);
+		if (!$user) {
+			$user = _elgg_services()->session->getLoggedInUser();
+		}
+
+		$return = $this->canEdit($user->guid);
```

Comment:
Also note get_entity() returns `false` while session->getLoggedInUser() may return `null`. We'd want to normalize this falsey value to whatever we generally pass to the permission hooks.